src/desktop/apps/fair/redirectFairRequests.ts

```diff
@@ -0,0 +1,32 @@
+import { NextFunction } from "express"
+import { ArtsyRequest, ArtsyResponse } from "lib/middleware/artsyExpress"
+
+export const redirectFairRequests = (
+  req: Partial<ArtsyRequest>,
+  res: Partial<ArtsyResponse>,
+  next: NextFunction
+): void => {
+  const isFairProfile = res.locals?.profile?.get("owner_type") === "Fair"
+
+  if (!isFairProfile) return next()
+
+  const fairSlug: string | null = res.locals.profile.get("owner")?.id
+
+  if (!fairSlug) {
+    return res.redirect(302, "/art-fairs")
+  }
+
+  const basePath = `/fair/${fairSlug}`
+
+  // this matches all /:id/info/* requests
+  if (req.route.path === "/:id/:tab*" && res.locals.tab === "info") {
+    return res.redirect(301, `/fair/${fairSlug}/info`)
+  }
+
+  // this matches /:id/browse/artworks, but not /:id/browse/exhibitors
+  if (req.route.path === "/:id/browse/*" && req.params["0"] === "artworks") {
+    return res.redirect(301, `/fair/${fairSlug}/artworks`)
+  }
+
+  return res.redirect(301, basePath)
+}
```

Comment:
Note that all new server-side code for our SPA-style apps [now lives here](https://github.com/artsy/force/blob/master/src/desktop/apps/artsy-v2/apps/). Redirects, etc, can live there; and here's a [couple](https://github.com/artsy/force/blob/master/src/desktop/apps/artsy-v2/apps/artwork/artworkMiddleware.tsx) [examples](https://github.com/artsy/force/blob/master/src/desktop/apps/artsy-v2/middleware/userRequiredMiddleware.tsx)

Reply:
I very much do not wish to try to figure out how to deal with the fall through redirects for the bare profile IDs routes at this moment. It seems like it'd be easier to deal with that once we don't have any other vanity-style routes.

Reply:
(But if you think it's not as painful as I'm imagining I can try tackling in a separate PR)

Reply:
No clue how painful it is, mostly just pointing out that we've moved a lot of this kind of logic over to the v2 folder. Np leaving it here for the tiime being, but might be worth adding a note of some sort at the top mentioning that the app has been moved into our app shell (or something).

Reply:
Yeah, good idea; will add a README in the app dir for now.

Reply:
even better 👍

Reply:
Oh, and I also see your original point (organization). For some reason I thought you were talking about moving the routes over to found. Will move!

Reply:
284b1a8dd41c419798030465446a607cc4bd18ce 🙇‍♂️